### PR TITLE
Batch Debugger.ScriptParsed events sent from the debug extension

### PR DESCRIPTION
**Pull request by annagrin**
Batch Debugger.ScriptParsed events sent from the debug extension to extension backend.

- batch extension events with method `Debugger.ScriptParsed` and send the batches every second to the server.
- add tests verifying that the extension backend receives the `Debugger.ScriptParsed` events
- fix flakes in debug_extension_tests and enable previously disabled tests

Note: We introduced sending `Debugger.ScriptParsed` events to fix the issue https://github.com/dart-lang/webdev/issues/1557, this PR is a follow up making the solution faster.

Closes: https://github.com/dart-lang/webdev/issues/1599

Related: https://github.com/dart-lang/webdev/issues/1621
Related: https://github.com/dart-lang/webdev/issues/1613

### CODE REVIEW

File: dwds/debug_extension/web/background.dart [hunk 1]
```diff
@@ -61,7 +62,10 @@ Tab _mostRecentDartTab;
 DebuggerTrigger _debuggerTrigger;
 
 class DebugSession {
-  final SocketClient socketClient;
+  // How often to send batched events.
```
Comment (elliette) [inline dwds/debug_extension/web/background.dart:65]: Nit: private fields should be listed after public ones

File: dwds/test/devtools_test.dart [hunk 2]
```diff
@@ -46,9 +46,7 @@ void main() {
     test('can launch devtools', () async {
       var windows = await context.webDriver.windows.toList();
       await context.webDriver.driver.switchTo.window(windows.last);
-      // TODO(grouma): switch back to `fixture.webdriver.title` when
-      // https://github.com/flutter/devtools/issues/2045 is fixed.
-      expect(await context.webDriver.pageSource, contains('Flutter'));
+      expect(await context.webDriver.title, contains('Dart DevTools'));
```
Comment (elliette) [inline dwds/test/devtools_test.dart:49]: This changes seem unrelated to this PR / the tests are still disabled. I think these changes should be made in a separate PR as part of re-enabling the tests.

Comment (annagrin) [reply dwds/test/devtools_test.dart:49]: agreed, will revert

Comment (annagrin) [reply dwds/test/devtools_test.dart]: Done!

File: dwds/CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,9 @@
 - Update `package:vm_service` to 8.3.0.
 - Convert JavaScript stack traces in uncaught exceptions to Dart stack traces.
 - Fix failure to set breakpoints on windows with a base change in index.html.
-
```
Comment (elliette) [inline dwds/CHANGELOG.md:6]: One more nit: Looks like the DWDS change is actually just moving `batched_stream.dart` into shared utilities. Let's update CHANGELOG to reflect that

**Review by elliette — APPROVED** [review]
One requested change than LGTM

File: dwds/CHANGELOG.md (hunk 3 above)
Comment (annagrin) [reply dwds/CHANGELOG.md:6]: done!
